src/app/app.component.ts

```diff
@@ -440,19 +443,19 @@ export class ArlasWuiComponent implements OnInit, AfterViewInit {
   public updateVisibleItems() {
     const idFieldName = this.collectionToDescription.get(this.previewListContrib.collection).id_path;
     setTimeout(() => {
-      this.visibleItems = this.previewListContrib.data.map(i => i.get(idFieldName))
+      const visibleItems = this.previewListContrib.data.map(i => i.get(idFieldName).toString())
         .filter(i => this.isElementInViewport(document.getElementById(i.toString())));
-      this.updateMapStyle(this.visibleItems);
-    }, 200);
+      this.updateMapStyle(visibleItems, this.previewListContrib.collection);
+    }, 1000);
```

Comment:
1 second seems to be a long delay

Reply:
I totally agree, I'm trying to figure out a proper way to update the map style when the list is completly loaded

Reply:
I reduced it